modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,449 @@
+                lp.set("pad_mode", "VALID");
+            }
+            else
+                CV_Error(Error::StsError, "Unsupported padding mode " + attribute_proto.s());
+        }
+        else if(attribute_name == "dilations")
+        {
+            CV_Assert(attribute_proto.ints_size() == 2);
+            lp.set("dilation_h",  attribute_proto.ints(0));
+            lp.set("dilation_w",  attribute_proto.ints(1));
+        }
+        else if (attribute_proto.has_i())
+        {
+            lp.set(attribute_name, attribute_proto.i());
+        }
+        else if (attribute_proto.has_f())
+        {
+            lp.set(attribute_name, attribute_proto.f());
+        }
+        else if (attribute_proto.has_s())
+        {
+            lp.set(attribute_name, attribute_proto.s());
+        }
+        else if (attribute_proto.floats_size() > 0)
+        {
+            lp.set(attribute_name, DictValue::arrayReal(
+                (float*)attribute_proto.mutable_floats(), attribute_proto.floats_size()));
+        }
+        else if (attribute_proto.ints_size() > 0)
+        {
+                lp.set(attribute_proto.name(), DictValue::arrayInt(
+                    (int*)attribute_proto.mutable_ints(), attribute_proto.ints_size()));
+        }
+        else if (attribute_proto.has_t() || attribute_proto.has_g() ||
+                    attribute_proto.strings_size() > 0 ||
+                        attribute_proto.tensors_size() > 0 ||
+                            attribute_proto.graphs_size() > 0)
+        {
+                CV_Error(Error::StsNotImplemented, "Unexpected attribute type");
+        }
+        else
+            CV_Error(Error::StsNotImplemented, "Unsupported attribute type");
+    }
+    return lp;
+}
+
+Mat ONNXImporter::getBlob(const opencv_onnx::NodeProto& node_proto,
+                    const std::map<std::string, Mat>& constBlobs, int index)
+{
+    std::map<std::string, Mat>::const_iterator constBlob;
+    constBlob = constBlobs.find(node_proto.input(index) );
+    if (constBlob == constBlobs.end()) {
+        CV_Error(Error::StsObjectNotFound,
+             "Blob " + node_proto.input(index) + " not found in const blobs");
+    }
+    return constBlob->second;
+}
+
+void ONNXImporter::populateNet(Net dstNet)
+{
+    CV_Assert(model_proto.has_graph());
+    opencv_onnx::GraphProto graph_proto = model_proto.graph();
+    std::map<std::string, Mat> constBlobs = getGraphTensors(graph_proto);
+
+    std::string framework_name;
+    if (model_proto.has_producer_name()) {
+        framework_name = model_proto.producer_name();
+    }
+
+    // create map with network inputs (without const blobs)
+    std::map<std::string, int> layer_id;
+    std::map<std::string, int>::iterator layerId;
+
+    int count = 0;
+    for (int j = 0; j < graph_proto.input_size(); j++)
+    {
+        if (constBlobs.find(graph_proto.input(j).name()) == constBlobs.end()) {
+            layer_id.insert(std::make_pair(graph_proto.input(j).name(), count));
```

Comment:
We need to create a test with a network which has more than one input. In example, element-wise summation of two inputs.
Please also add a comment above what this loop does.